src/idl/include/idl/tree.h

```diff
@@ -341,6 +346,14 @@ struct idl_struct {
   idl_extensibility_t extensibility;
 };
 
+typedef struct idl_forward_decl idl_forward_decl_t;
+struct idl_forward_decl {
+  idl_node_t node;
+  struct idl_name *name;
+  struct idl_scoped_name *scoped_name;
+  idl_mask_t type_mask;
```

Comment:
Actually, the idea behind the mask in the node is to make combinations. Like a literal value is `IDL_LITERAL` + valid type, I intended the the forward declarations to use `IDL_FORWARD` + (`IDL_STRUCT` or `IDL_UNION`). Is there a strong reason not to do it like that?

Reply:
No, there's not, I've changed this in c80f50d21d8712247a7a6c7c48aae7823ce6dc0a